src/requests/export.ts

```diff
@@ -0,0 +1,69 @@
+import { GraphQLClient } from "graphql-request";
+import { getSessionById, getSessionPassport } from "./session";
+import { findPublisedFlowBySessionId, getFlowName } from "./flow";
+import { computeCSVData } from "../export/csv";
+import { computeBOPSParams } from "../export/bops";
+import type { CSVData } from "../export/csv/model";
+import type { BOPSFullPayload } from "../export/bops/model";
+
+export class ExportClient {
+  protected client: GraphQLClient;
+
+  constructor(client: GraphQLClient) {
+    this.client = client;
+  }
+
+  csvData(sessionId: string): Promise<CSVData> {
+    return generateCSVData(this.client, sessionId);
+  }
+
+  bopsPayload(sessionId: string): Promise<BOPSFullPayload> {
+    return generateBOPSPayload(this.client, sessionId);
+  }
+}
+
+export async function generateCSVData(
+  client: GraphQLClient,
+  sessionId: string
+): Promise<CSVData> {
+  const bopsData = await generateBOPSPayload(client, sessionId);
+  if (!bopsData) {
+    throw new Error(
+      `Cannot fetch BOPS Params for session ${sessionId} so Cannot generate CSV Data`
+    );
+  }
+  const passport = await getSessionPassport(client, sessionId);
+  if (!passport) {
+    throw new Error(
+      `Cannot find passport for session ${sessionId} so Cannot generate CSV Data`
+    );
+  }
+  return computeCSVData({
+    sessionId,
+    bopsData,
+    passport,
+  });
+}
+
+export async function generateBOPSPayload(
+  client: GraphQLClient,
+  sessionId: string
+): Promise<BOPSFullPayload> {
+  try {
+    const session = await getSessionById(client, sessionId);
+    if (!session) throw new Error(`Cannot find session ${sessionId}`);
+    const flow = await findPublisedFlowBySessionId(client, sessionId);
```

Comment:
This is a functional change which I think we want. `findPublishedFlowBySessionId` is looking up the flow and calling `getMostRecentPublishedFlowBeforeTimestamp` to get the flow as it was published when the session was last updated. It should mean that the export can happen at any time but always points to the flow as it was when the session was completed.

Reply:
This is a good catch and I agree is correct - without this we could have different sections, different passport vars, etc which would impact the representation of the user's journey.